colorbleed/plugins/maya/publish/collect_renderlayers.py

```diff
@@ -62,43 +78,44 @@ def process(self, context):
 
             # Include (optional) global settings
             # TODO(marcus): Take into account layer overrides
-            try:
-                avalon_globals = maya.lsattr("id", "avalon.renderglobals")[0]
-            except IndexError:
-                pass
-            else:
-                _globals = maya.read(avalon_globals)
-                data["renderGlobals"] = self.get_global_overrides(_globals)
+            # Get global overrides and translate to Deadline values
+            overrides = self.translate_overrides(render_globals)
+            data.update(**overrides)
 
             instance = context.create_instance(layername)
             instance.data.update(data)
 
     def get_render_attribute(self, attr):
         return cmds.getAttr("defaultRenderGlobals.{}".format(attr))
 
-    def get_global_overrides(self, globals):
-        """
-        Get all overrides with a value, skip those without
+    def translate_overrides(self, render_globals):
```

Comment:
Maybe this reads better when called `parse_options`? or `parse_overrides`?

Reply:
Here's another draft that might inspire to reformat some code:
```python
    def parse_options(self, render_globals):
        """Get all overrides with a value, skip those without

        Here's the kicker. These globals override defaults in the submission
        integrator, but an empty value means no overriding is made.
        Otherwise, Frames would override the default frames set under globals.

        Args:
            render_globals (str): collection of render globals

        Returns:
            dict: only overrides with values
        """

        attributes = maya.read(render_globals)

        options = {"renderGlobals": {}}
        
        options['renderGlobals']['Priority'] = attributes['priority']
        
        # Machine list
        machine_list = attributes["machineList"]
        if machine_list:
            key = "Whitelist" if  attributes["whitelist"] else "Blacklist"
            options['renderGlobals'][key] = machine_list
        
        # Suspend publish job
        state = "Suspended" if attributes["suspendPublishJob"] else "Active" 
        options["suspendPublishJob"] = state 

        return options
```